src/screens/login/components/functions/oidc.ts

```diff
@@ -0,0 +1,177 @@
+    const onboardUrl = oidcUrl.searchParams.get('onboard_url')
+
+    if (!code || !onboardUrl) {
+      return null
+    }
+
+    const onboardUrlWithRedirect = new URL(onboardUrl)
+    onboardUrlWithRedirect.searchParams.append(
+      OIDC_ONBOARD_CALLBACK_URL_PARAM,
+      OIDC_ONBOARD_CALLBACK_URL
+    )
+
+    return { code, onboardUrl: onboardUrlWithRedirect.toString() }
+  } catch (error) {
+    const errorMessage = getErrorMessage(error)
+    log.error(
+      `Something went wrong while trying to extract OIDC onboarding info from URL: ${errorMessage}`
+    )
+    return null
+  }
+}
+
+const parseOIDCLoginUrl = (url: string): OidcLoginCallback | null => {
+  try {
+    const oidcUrl = new URL(url)
+
+    const code = oidcUrl.searchParams.get('code')
+    const fqdn = oidcUrl.searchParams.get('fqdn')
+    const defaultRedirection = oidcUrl.searchParams.get('default_redirection')
+
+    if (!code || !fqdn) {
+      return null
+    }
+
+    return { code, fqdn, defaultRedirection }
+  } catch (error) {
+    const errorMessage = getErrorMessage(error)
+    log.error(
+      `Something went wrong while trying to extract OIDC login info from URL: ${errorMessage}`
+    )
+    return null
+  }
+}
+
+const parseOIDCResultUrl = (url: string): OidcCallback | null => {
+  return parseOIDCLoginUrl(url) ?? parseOIDCOnboardingStartUrl(url)
+}
+
+/**
+ * Display InAppBrowser to the given OIDC url
+ * @param request - NavigationRequest called by the Cloudery to start the OIDC process
+ * @returns OIDC result
+ */
+export const processOIDC = (
+  request: WebViewNavigation
+): Promise<OidcCallback> => {
+  return new Promise((resolve, reject) => {
+    const oidcUrl = new URL(request.url)
+    oidcUrl.searchParams.append(OIDC_CALLBACK_URL_PARAM, OIDC_CALLBACK_URL)
+
+    void showInAppBrowser({ url: oidcUrl.toString() }).then(result => {
+      if (result.type === 'cancel') {
+        reject(USER_CANCELED)
+      }
+      return
+    })
+
+    const subscription = Linking.addEventListener('url', ({ url }) => {
+      const oidcResult = parseOIDCResultUrl(url)
+
+      if (!oidcResult) {
+        return reject(INVALID_CALLBACK)
+      }
+
+      resolve(oidcResult)
+      subscription.remove()
+      void closeInAppBrowser()
+    })
+  })
+}
+
+/**
+ * Extract created instance and onboarded_redirection from NavigationRequest
+ * @param request - NavigationRequest called by the Cloudery after Onboarding has been finished
+ * @returns Created instance info
+ */
+export const parseOidcOnboardingFinishedUrl = (
+  request: WebViewNavigation
+): OidcOnboardingEndCallback | null => {
+  try {
+    const oidcUrl = new URL(request.url)
+
+    const fqdn = oidcUrl.searchParams.get('fqdn')
+    const onboardedRedirection = oidcUrl.searchParams.get(
+      'onboarded_redirection'
+    )
+
+    if (!fqdn) {
+      return null
+    }
+
+    return { fqdn, onboardedRedirection }
+  } catch (error) {
+    const errorMessage = getErrorMessage(error)
+    log.error(
```

Comment:
same here

Reply:
Fixed in https://github.com/cozy/cozy-flagship-app/compare/c351d6a960edac9811994f000ba5a790dcd992bf..02727db31d8c486e4afd0241610cd83f87ebf22a